### PR TITLE
feat: AWS Inspector Integration with C1WS

### PR DESCRIPTION
**Pull request by wschadrach**
## Cloud One Service

[X] **Workload Security**

## Proposed Changes
  - Add AWS Inspector Integration with C1WS

### CODE REVIEW

**Review by felipecosta09 — CHANGES_REQUESTED** [review]
We need to have the code out of the zip file to run the Linter and security checks appropriately.

For the code to be used in the lambda, I think it would be best to have an s3 bucket and upload the code via cloud formation, but I'm open to other options as well, check the File Storage Plugins because they explore some deployment options there that might be valuable:

<https://github.com/trendmicro/cloudone-filestorage-plugins/blob/master/post-scan-actions/aws-python-promote-or-quarantine/other-installation-methods.md>

File: Workload-Security/Integration/aws-python-amazon-inspector-integration/README.md [hunk 1]
```diff
@@ -0,0 +1,24 @@
...
+
+The goal of this integration is to be able to show Trend Micro customers that are using Amazon Inspector what the Amazon findings (CVEs) are being protected by IPS rules. In order to achieve this goal, we may need to let customers understand the security risk / vulnerability issue in their environment, then we could have some opportunity to adopt a C1WS solution to mitigate / solve their issue using the ips rules.
+This deployment will help customer in need or that are using Inspector to better link both cloud one Workload Security and Amazon Inspector.
+The functionality of the scripts
+
+In this project we have two deployments. The first deployment (vulnerability report) is responsible to send out a report of the vulnerabilities that are being protected by ips rules among the findings (CVEs) Inspector is detecting. The second deployment (auto assign ips rule) is responsible for assigning the ips rules found based on the Inspector findings to the computers (EC2 instances). This means if the Inspector detected any findings (CVEs) then the script will send information to Trend Micro Workload Security who will protect the instances from whatever was detected?
+Deployment
+
```
Comment (JustinDPerkins) [inline Workload-Security/Integration/aws-python-amazon-inspector-integration/README.md:16]: I would add more details to the readme page. Ensure that all prereqs are listed before hand. Such as SES set up. Cloud One Api Key generation.

File: Workload-Security/Integration/aws-python-amazon-inspector-integration/auto-assign-ips-rule.yaml [hunk 2]
```diff
@@ -0,0 +1,106 @@
+Parameters:
+    Region:
+      Type: String
+      Description: Please enter the region in which the lambda will be executed
+    s3Bucket:
+      Type: String
+      Description: Enter the name of the s3 bucket where you upload the Lambda Layer and package
+    ApiKey:
```
Comment (JustinDPerkins) [inline Workload-Security/Integration/aws-python-amazon-inspector-integration/auto-assign-ips-rule.yaml:8]: This needs to be secured. Not in plain-view

File: Workload-Security/Integration/aws-python-amazon-inspector-integration/auto-assign-ips-rule.yaml [hunk 3]
```diff
@@ -0,0 +1,106 @@
...
+      Type: String
+      Description: Email id to be used for sending report
+    EmailRecipients:
+      Type: String
+      Description: Comma separated email ids for recieving report
+    WorkloadSecurityHost:
+      Type: String
+      Description: Enter your Workload Security Host 
+      Default: https://workload.trend-us-1.cloudone.trendmicro.com/api
+    CronSchedule:
```
Comment (JustinDPerkins) [inline Workload-Security/Integration/aws-python-amazon-inspector-integration/auto-assign-ips-rule.yaml:21]: Maybe include a default cron value of weekly reports.

**Review by JustinDPerkins — CHANGES_REQUESTED** [review]

**Comment by felipecosta09** [comment]
Hey @wschadrach any updates?

**Comment by wschadrach** [comment]
> Hey @wschadrach any updates?

Yes, will repost the updated files soon

**Comment by wschadrach** [comment]
> We need to have the code out of the zip file to run the Linter and security checks appropriately.
> 
> For the code to be used in the lambda, I think it would be best to have an s3 bucket and upload the code via cloud formation, but I'm open to other options as well, check the File Storage Plugins because they explore some deployment options there that might be valuable:
> 
> https://github.com/trendmicro/cloudone-filestorage-plugins/blob/master/post-scan-actions/aws-python-promote-or-quarantine/other-installation-methods.md

**Pull request closed without merging**